src/main/groovy/org/transmartproject/batch/concept/ConceptTree.groovy

```diff
@@ -54,10 +54,21 @@ class ConceptTree {
         }
     }
 
+    //TODO We should have just one way to check is record new @{see getConceptNodesToInsert}
     boolean isNew(ConceptNode node) {
         !(node.path in oldConceptPaths)
     }
 
+    Collection<ConceptNode> getNewConceptNodes() {
+        allConceptNodes.findAll {
+            !(it.path in oldConceptPaths)
+        }
+    }
+
+    Collection<ConceptNode> getConceptNodesToInsert() {
+        nodeMap.values().findAll { !it.code }
+    }
```

Comment:
I don't get the need for both this one and `getNewConceptNodes` besides that this one cannot be called after the ids have been reserved (which makes it dangerous).